File: 50dias/cpp/25.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

const int PONTUACAO_BASE = 3;

int placar = 0;
int mult_pontos = 1;
int multiplicando = 0;
int multiplicador = 0;


void InicializaPlacar() {
    mult_pontos = 1;
    placar = 0;
}

void AtualizaPontos(int nivel) {
    placar += mult_pontos * PONTUACAO_BASE;
    if ((nivel % 5) == 0)
        mult_pontos++;
}

int GeraNumeroEntre1e10() {
    // rand() % 10 gera um numero entre 0 e 9
    return (rand() % 10) + 1;
}

bool EfetuaQuestao(int multiplicando, int multiplicador) {
    int resposta;
    cout << "\t";
    cout << multiplicando << "x" << multiplicador << " = ";
    cin >> resposta;

    return resposta == multiplicador * multiplicando;
}

void ImprimeCabecalho(int nivel) {
    system("cls");
    cout << "-------------------------------------------------" << endl;
    cout << " Nivel: " << nivel << "   |   Placar: " << placar;
    cout << "   |   Multiplicador: " << mult_pontos << endl;
    cout << "-------------------------------------------------" << endl;
}

void EfetuaQuestoes() {
    int nivel = 0;
    bool continuar = true;
    do {
        nivel++;

        multiplicando = GeraNumeroEntre1e10();
        multiplicador = GeraNumeroEntre1e10();

        ImprimeCabecalho(nivel);
        continuar = EfetuaQuestao(multiplicando,
            multiplicador);
        if (continuar)
            AtualizaPontos(nivel);
    } while (continuar);
}

void UsuarioErrou() {
    cout << "Infelizmente voce errou!" << endl;
    cout << "O resultado de " << multiplicando << "x";
    cout << multiplicador << " = " << multiplicando * multiplicador;
    cout << endl << endl;
}

bool UsuarioQuerContinuar() {
    char tentarNovamente;
    cout << "Voce quer tentar novamente? [s]im/[n]ao : ";
    cin >> tentarNovamente;
    return tentarNovamente == 's';
}

int main()
{
    srand(time(0));
    do {
        InicializaPlacar();
        EfetuaQuestoes();
        UsuarioErrou();
    } while (UsuarioQuerContinuar());

    return 0;
}

File: 50dias/python/25.py
import random
import os

PONTUACAO_BASE = 3

placar = 0
mult_pontos = 1
multiplicando = 0
multiplicador = 0

def InicializaPlacar():
    global mult_pontos, placar
    mult_pontos = 1
    placar = 0

def AtualizaPontos(nivel):
    global mult_pontos, placar
    placar += mult_pontos * PONTUACAO_BASE
    if (nivel % 5) == 0:
        mult_pontos += 1

def GeraNumeroEntre1e10():
    return random.randint(1, 10)

def EfetuaQuestao():
    resposta = int(input(str(multiplicando) + 'x'+str(multiplicador)+' = '))
    return resposta == multiplicando*multiplicador

def ImprimeCabecalho(nivel):
    os.system('cls')
    cabecalho = "---------------------------------------------------------\n" \
                " Nivel: " + str(nivel) +"   |   Placar: " + str(placar) + \
                "   |   Multiplicador: " + str(mult_pontos) + "\n"\
                "---------------------------------------------------------\n"
    cabecalho.format(nivel=nivel,
                     placar=placar,
                     mult=mult_pontos)
    print(cabecalho)

def EfetuaQuestoes():
    global multiplicando, multiplicador
    nivel = 0
    continuar = True
    while continuar:
        nivel += 1
        multiplicando = GeraNumeroEntre1e10()
        multiplicador = GeraNumeroEntre1e10()

        ImprimeCabecalho(nivel)
        continuar = EfetuaQuestao()
        if continuar:
            AtualizaPontos(nivel)


def UsuarioErrou():
    print('Infelizmente voce errou!\n')
    print('O resultado de {prim}x{sec} = {ter}'.format(prim=multiplicando,
                                                       sec=multiplicador,
                                                       ter=multiplicando*multiplicador))
    print('\n\n')

def UsuarioQuerContinuar():
    tentar = input('Voce quer tentar novamente? [s]im/[n]ao: ')
    return tentar == 's'

def main():
    continuar = True

    while continuar:
        InicializaPlacar()
        EfetuaQuestoes()
        UsuarioErrou()
        continuar = UsuarioQuerContinuar()


main()